pkg/pod/ttlDestroyHandler.go

```diff
@@ -56,12 +58,18 @@ func (t ttlDestroyHandler) Handle(pod v1.Pod) error {
 	}
 
 	if t.ttlBeforeNow(creationTimestamp, ttl) {
-		id, err := strconv.ParseUint(pod.Labels["im-id"], 10, 64)
+		id, err := strconv.ParseUint(pod.Labels["im-instance-id"], 10, 64)
 		if err != nil {
 			return err
 		}
+
 		payload := struct{ ID uint }{uint(id)}
-		t.producer.Produce(ttlDestroy, payload)
+		correlationId := uuid.NewString()
```

Comment:
I would create the id at the top. Then do https://pkg.go.dev/log/slog#With `logger=logger.With("correlationId", correlationID)` so that every log line has the attribute.